lib/asciidoctor/converter/semantic-html5.rb

```diff
@@ -40,11 +42,47 @@ def generate_header node
       if (doctitle = generate_document_title node)
         result << doctitle
       end
+      if (revision = generate_revision node)
+        result << revision
+      end
       result << '</header>'
       result.join LF
     end
   end
 
+  def generate_revision node
+    return unless (node.attr? 'revnumber') || (node.attr? 'revdate') || (node.attr? 'revremark')
+
+    header = []
+    body = []
+    if (revnumber = node.attr 'revnumber')
+      header << '<th>Version</th>'
+      body << %(<td data-title="#{node.attr 'version-label', ''}">#{revnumber}</td>)
+    end
+    if (revdate = node.attr 'revdate')
+      header << '<th>Date</th>'
+      date = ::Date._parse revdate
+      date_text = "#{date[:year]}-#{date[:mon].to_s.rjust 2, '0'}-#{date[:mday].to_s.rjust 2, '0'}"
+      body << %(<td data-title="Date"><time datetime="#{date_text}">#{revdate}</time></td>)
+    end
+    if (revremark = node.attr 'revremark')
+      header << '<th>Remark</th>' if node.attr? 'revremark'
+      body << %(<td data-title="Remark">#{revremark}</td>)
+    end
+    %(<table class="revision">
```

Comment:
I quite like the table structure.
One drawback is that the number of columns is flexible so we cannot use the column index to apply a specific style. In other words, we don't know for sure what the first column contains.
In this context, it could be useful to add a class or a data attribute to identify columns.

Do you think we should always generate all columns (i.e., use a fixed number of columns)?

Reply:
One thing we lose by using the table is the concise byline we see here: https://beanvalidation.org/2.0/spec/ and here: https://jakarta.ee/specifications/cdi/3.0/jakarta-cdi-spec-3.0.html. Clearly, a revision table is also desired in both of those cases.

We also run up against the case that it's only possible to specify a single revision. Though perhaps this converter is making room for that eventuality. Still, I wonder if the table is something that can be turned on using an attribute like `revtable`, or `:revstyle: table`.

Reply:
> One thing we lose by using the table is the concise byline we see here: beanvalidation.org/2.0/spec and here: jakarta.ee/specifications/cdi/3.0/jakarta-cdi-spec-3.0.html. 

It's possible to display the table as a single line using a few lines of CSS:

```html
<table class="revision">
<thead>
<tr>
<th>Version</th>
<th>Date</th>
<th>Remark</th>
</tr>
</thead>
<tbody>
<tr>
<td data-title="Version">1.2.3</td>
<td data-title="Date"><time datetime="2021-10-31">October 31, 2021</time></td>
<td>Fall incarnation<td>
</tr>
</tbody>
</table>
```

```css
table.revision > thead {
  display: none;
}
table.revision tr > td:first-child::before {
  content: attr(data-title) ' '
}
table.revision tr > td:first-child::after {
  content: ','
}
table.revision tr > td:nth-child(3):not(:empty)::before {
  content: ' |'
}
```

And here's the result:

![111903820-4847b400-8a44-11eb-8229-dc5be80635c3](https://user-images.githubusercontent.com/333276/115141545-453ce500-a03d-11eb-9cda-d2673f7cd725.png)


With a few extra lines of CSS, it's also possible to support multiple revisions:


```css
table.revision tbody > tr {
  display: inline-flex;
}
table.revision tbody > tr:not(:first-child)::before {
  content: ' - '
}
```

![multi-revs](https://user-images.githubusercontent.com/333276/115141939-4f5fe300-a03f-11eb-8dd0-8c35a612578c.png)


> We also run up against the case that it's only possible to specify a single revision. Though perhaps this converter is making room for that eventuality. Still, I wonder if the table is something that can be turned on using an attribute like revtable, or :revstyle: table.

That's a good idea.
If you have more than one revision you will probably want to display them as a table.

We could also configure the location of the revision(s) table (similar to what we do with the Table Of Contents):

- `preamble`
- `macro`
- `header` (default) 


```
:revstyle: table
:revision: preamble
:revision-title: Revisions
```

![demo-tablestyle-revs](https://user-images.githubusercontent.com/333276/115142314-669fd000-a041-11eb-8b29-e46ca731625e.png)

Even if we won't implement that in the near future, it seems flexible enough to accommodate "advanced" features.

Reply:
> Do you think we should always generate all columns (i.e., use a fixed number of columns)?

I think we should, it will make things easier and we can use `:not(:empty)` to style accordingly.

Reply:
> > Do you think we should always generate all columns (i.e., use a fixed number of columns)?

> I think we should, it will make things easier and we can use :not(:empty) to style accordingly.

Agreed. Go for it.

Reply:
> We could also configure the location of the revision(s) table (similar to what we do with the Table Of Contents):

That's a great idea. If it feels out of scope, I'm fine with making it a separate issue.

Based on what you showed, I'm prepared to move forward with the table as you have it (with the fixed number of columns).

Reply:
I'm glad you like it!
Yes, let's do that as a follow-up issue 😃